File: codes/c++_codes/primeSieve.cpp
#include<bits/stdc++.h>
using namespace std;
void primeSieve(int n){
    int i,j;
    vector<bool> p(n+1);
    fill(p.begin(),p.end(),true);
    p[1]=p[0]=false;
    for(i=2;i*i<=n;i++){
        if(p[i]){
            for(j=i*i;j<=n;j+=i){
                p[j]=false;
            }
        }
    }
    cout<<"The prime numbers in the range are:\n";
    for(i=2;i<=n;i++){
        if(p[i]){
            cout<<i<<" ";
        }
    }
    return;
}
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    int n;
    cout<<"Enter the upper limit for primes: ";
    cin>>n;
    primeSieve(n);
    return 0;
}

File: codes/java_codes/primeSieve.java
import java.util.*;
public class primeSieve {
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        int n;
        System.out.println("Enter the range of prime: ");
        n=sc.nextInt();
        prime(n);
    }
    public static void prime(int n){
        boolean[] p=new boolean[n+1];
        for(int i=2;i<=n;i++){
            p[i]=true;
        }
        p[0]=p[1]=false;
        for(int i=2;i*i<=n;i++){
            if(p[i]){
                for(int j=i*i;j<=n;j+=i){
                    p[j]=false;
                }
            }
        }
        System.out.println("The prime numbers are: ");
        for(int i=2;i<=n;i++){
            if(p[i]){
                System.out.print(i+" ");
            }
        }
    }
}